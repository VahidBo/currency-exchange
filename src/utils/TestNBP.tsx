import React, { PropsWithChildren } from 'react';
import { NativeBaseProvider } from 'native-base';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const TestNBP = ({ children }: PropsWithChildren<Record<string, never>>) => (
  <NativeBaseProvider initialWindowMetrics={inset}>{children}</NativeBaseProvider>
);
