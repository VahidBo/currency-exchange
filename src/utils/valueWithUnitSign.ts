import { Currency } from 'src/components';

export function valueWithUnitSign(value: number, currency: Currency): string {
  return { USD: `$${value}`, EUR: `€${value}`, GBP: `£${value}` }[currency];
}
