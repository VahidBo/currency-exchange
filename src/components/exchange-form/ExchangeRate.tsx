import React from 'react';
import { Currency } from './types';
import { Badge, Box, Divider, Spinner } from 'native-base';
import { StyleSheet } from 'react-native';
import { valueWithUnitSign } from '../../utils';

export interface ExchangeRateProps {
  rate: number | null;
  fromCurrency: Currency;
  toCurrency: Currency;
}

export function ExchangeRate({ rate, fromCurrency, toCurrency }: ExchangeRateProps) {
  return (
    <Box height={100} width="100%" display="flex" alignItems="center" justifyContent="center">
      <Divider my="10" bg="indigo.500" />
      <Badge
        colorScheme="indigo.500"
        variant="outline"
        style={styles.badge}
        _text={{ fontSize: 14, color: 'indigo.500' }}
      >
        {rate ? (
          <>{`${valueWithUnitSign(1, fromCurrency)} = ${valueWithUnitSign(rate, toCurrency)}`}</>
        ) : (
          <Spinner color="indigo.500" />
        )}
      </Badge>
    </Box>
  );
}

const styles = StyleSheet.create({
  badge: {
    position: 'absolute',
    backgroundColor: 'white',
    borderRadius: 14,
    minWidth: '40%',
  },
});
