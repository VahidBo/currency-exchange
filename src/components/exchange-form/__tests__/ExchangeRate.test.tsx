import React from 'react';
import renderer from 'react-test-renderer';
import { ExchangeRate } from '../ExchangeRate';
import { TestNBP } from '../../../utils/TestNBP';

test('Snapshot of ExchangeRate', () => {
  const tree = renderer
    .create(
      <TestNBP>
        <ExchangeRate rate={0.23422} toCurrency="EUR" fromCurrency="GBP" />
      </TestNBP>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Snapshot of ExchangeRate in loading state', () => {
  const tree = renderer
    .create(
      <TestNBP>
        <ExchangeRate rate={null} toCurrency="EUR" fromCurrency="GBP" />
      </TestNBP>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
