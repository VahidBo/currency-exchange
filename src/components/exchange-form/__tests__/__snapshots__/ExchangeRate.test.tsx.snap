// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of ExchangeRate 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 100,
        "justifyContent": "center",
        "marginBottom": 16,
        "width": "100%",
      }
    }
  >
    <View
      aria-orientation="horizontal"
      style={
        Object {
          "backgroundColor": "#6366f1",
          "height": 1,
          "marginBottom": 40,
          "marginTop": 40,
          "width": "100%",
        }
      }
      thickness="1"
    />
    <View
      colorScheme="indigo.500"
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#06b6d4",
            "borderRadius": 2,
            "borderWidth": 1,
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
          Object {
            "backgroundColor": "white",
            "borderRadius": 14,
            "minWidth": "40%",
            "position": "absolute",
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#6366f1",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "500",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        £1 = €0.23422
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Snapshot of ExchangeRate in loading state 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": 100,
        "justifyContent": "center",
        "marginBottom": 16,
        "width": "100%",
      }
    }
  >
    <View
      aria-orientation="horizontal"
      style={
        Object {
          "backgroundColor": "#6366f1",
          "height": 1,
          "marginBottom": 40,
          "marginTop": 40,
          "width": "100%",
        }
      }
      thickness="1"
    />
    <View
      colorScheme="indigo.500"
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#06b6d4",
            "borderRadius": 2,
            "borderWidth": 1,
            "paddingBottom": 2,
            "paddingLeft": 8,
            "paddingRight": 8,
            "paddingTop": 2,
          },
          Object {
            "backgroundColor": "white",
            "borderRadius": 14,
            "minWidth": "40%",
            "position": "absolute",
          },
        ]
      }
    >
      <ActivityIndicator
        accessibilityLabel="loading"
        accessible={true}
        color="#6366f1"
        size="small"
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
